src/constraints/capacity.jl

```diff
@@ -139,22 +135,25 @@ function add_capacity_constraints!(
     # - Create capacity limit for incoming flows
     assets_profile_times_capacity_in =
         model[:assets_profile_times_capacity_in] = [
-            if row.asset ∈ Ai[row.year]
+            if row.asset ∈ decommissionable_assets_using_compact_method
                 @expression(
                     model,
-                    profile_aggregation(
-                        Statistics.mean,
-                        graph[row.asset].rep_periods_profiles,
-                        row.year,
-                        row.year,
-                        ("availability", row.rep_period),
-                        row.timesteps_block,
-                        1.0,
-                    ) * (
-                        graph[row.asset].capacity * (
-                            graph[row.asset].initial_units[row.year][row.year] +
-                            assets_investment[row.year, row.asset]
-                        )
+                    graph[row.asset].capacity * sum(
+                        profile_aggregation(
+                            Statistics.mean,
+                            graph[row.asset].rep_periods_profiles,
+                            row.year,
+                            v,
+                            ("availability", row.rep_period),
+                            row.timesteps_block,
+                            1.0,
+                        ) *
+                        accumulated_units_compact_method[accumulated_set_using_compact_method_lookup[(
+                            row.asset,
+                            row.year,
+                            v,
+                        )]] for v in V_all if
+                        (row.asset, row.year, v) in accumulated_set_using_compact_method
```

Comment:
This conditional is inside the `sum`, so it might be slow.
The solution would be defining something like

```
some_set[a, y] = [v for v in V_all if (a, y, v) in accumulated_set_using_compact_method]
```

And looping over `some_set[row.asset, row.year]`.
This might be affected by the refactor, but since this array of expression is already defined as an array of expressions looped over the rows of a table, maybe this code won't be improved simply by doing the refactor.

Reply:
Good point to consider in the refactor and then see how we tackle this. Thanks!